### PR TITLE
fixed some phpDocs & other details

### PR DESCRIPTION
**Pull request by vondrasekpavel**
Hi,

I fixed some phpDocs & changed some details.

Please check it because I'm groggy and some methods don't have tests :(

### CODE REVIEW

File: src/Jackalope/ObjectManager.php [hunk 1]
```diff
@@ -785,6 +801,7 @@ public function registerNodeTypesCnd($cnd, $allowUpdate)
                 $nsRegistry->registerNamespace($prefix, $uri);
             }
 
+            // TODO: Is @return type bool wrong?
```
Comment (dbu) [inline src/Jackalope/ObjectManager.php]: yes, the `@return` is wrong. this is called by NodeTypeManager::registerNodeTypesCnd which states its an \Iterator.

rather than put the whole description of that iterator here, we could do `@see NodeTypeManagerInterface::registerNodeTypesCnd`

File: src/Jackalope/Helper.php [hunk 2]
```diff
@@ -24,16 +24,15 @@ class Helper
...
     {
         if (! $node->hasAttribute($attribute)) {
             throw new RepositoryException("Expected attribute $attribute not found on ".$node->getNodePath());
         }
-        if ('false' === $node->getAttribute($attribute)) {
-            return false;
-        }
 
-        return true;
+        return !('false' === $node->getAttribute($attribute));
```
Comment (vondrasekpavel) [inline src/Jackalope/Helper.php]: yes, you are right :)

File: src/Jackalope/ImportExport/ImportExport.php [hunk 3]
```diff
@@ -49,7 +49,7 @@ class ImportExport implements ImportUUIDBehaviorInterface
      * @param boolean                    $skipBinary as in exportSystemView
      * @param boolean                    $noRecurse  as in exportSystemView
      *
-     * @see PHPCR\SessionInterface::exportSystemView
+     * @see \PHPCR\SessionInterface::exportSystemView
```
Comment (vondrasekpavel) [inline src/Jackalope/ImportExport/ImportExport.php]: ok

File: src/Jackalope/ImportExport/ImportExport.php [hunk 4]
```diff
@@ -66,7 +66,9 @@ public static function exportSystemView(NodeInterface $node, NamespaceRegistryIn
      * @param boolean                    $skipBinary as in exportDocumentView
      * @param boolean                    $noRecurse  as in exportDocumentView
      *
-     * @see PHPCR\SessionInterface::exportDocumentView
+     * @throws RepositoryException
+     *
+     * @see \PHPCR\SessionInterface::exportDocumentView
```
Comment (vondrasekpavel) [inline src/Jackalope/ImportExport/ImportExport.php]: ok

File: src/Jackalope/ImportExport/ImportExport.php [hunk 5]
```diff
@@ -244,7 +254,7 @@ private static function addNode(NodeInterface $parentNode, $nodename, $type, $pr
             $type = 'nt:unstructured';
         }
 
-        if ('jcr:root' == $nodename && isset($existing) && $existing->getDepth() === 0 && self::IMPORT_UUID_COLLISION_REPLACE_EXISTING == $uuidBehavior) {
+        if ('jcr:root' === $nodename && isset($existing) && self::IMPORT_UUID_COLLISION_REPLACE_EXISTING === $uuidBehavior && $existing->getDepth() === 0) {
```
Comment (vondrasekpavel) [inline src/Jackalope/ImportExport/ImportExport.php]: ok

File: src/Jackalope/ImportExport/ImportExport.php [hunk 6]
```diff
@@ -320,11 +334,12 @@ private static function exportSystemViewRecursive(NodeInterface $node, Namespace
 
         foreach ($node->getProperties() as $name => $property) {
             /** @var $property \PHPCR\PropertyInterface */
-            if ($name == 'jcr:primaryType' || $name == 'jcr:mixinTypes' || $name == 'jcr:uuid') {
+
+            if ($name === 'jcr:primaryType' || $name === 'jcr:mixinTypes' || $name === 'jcr:uuid') {
```
Comment (vondrasekpavel) [inline src/Jackalope/ImportExport/ImportExport.php]: ok

File: src/Jackalope/Lock/Lock.php [hunk 7]
```diff
@@ -90,6 +92,10 @@ public function setIsDeep($isDeep)
      * {@inheritDoc}
      *
      * @api
+     *
+     * @throws NotImplementedException
```
Comment (vondrasekpavel) [inline src/Jackalope/Lock/Lock.php]: ok, I removed it :)

File: src/Jackalope/Node.php [hunk 8]
```diff
@@ -597,7 +598,7 @@ public function getNode($relPath)
     {
         $this->checkState();
 
-        if (strlen($relPath) == 0 || '/' == $relPath[0]) {
+        if ($relPath === '' || '/' === $relPath[0]) {
```
Comment (vondrasekpavel) [inline src/Jackalope/Node.php]: ok

File: src/Jackalope/ObjectManager.php [hunk 9]
```diff
@@ -779,9 +781,17 @@ protected function pathArrayToPropertiesIterator($propertyPaths)
      *
      * @param  string  $cnd         a string with cnd information
      * @param  boolean $allowUpdate whether to fail if node already exists or to update it
-     * @return bool    true on success
+     * @return bool|\Iterator    true on success or \Iterator over the registered node types if repository is not able to process
```
Comment (dbu) [inline src/Jackalope/ObjectManager.php]: i think this is not correct. this method always registers the node types, either directly through transport or by parsing the file and calling the node type manager registerNodeTypes method.

if neither is possible, it throws UnsupportedRepositoryOperationException

so no boolean is ever returned.

Comment (vondrasekpavel) [inline src/Jackalope/ObjectManager.php]: Ok, think we must repair interface because bool return type is defined here:

snimek obrazovky 2016-09-23 v 14 46 11

File: src/Jackalope/Transport/AbstractReadLoggingWrapper.php [hunk 10]
```diff
@@ -191,7 +191,7 @@ public function setNodeTypeManager($nodeTypeManager)
     /**
      * {@inheritDoc}
      */
-    public function getNodeTypes($nodeTypes = array())
+    public function getNodeTypes(array $nodeTypes = array())
```
Comment (dbu) [inline src/Jackalope/Transport/AbstractReadLoggingWrapper.php]: i had to revert this change to the interface you did in the first PR, as its a BC break for the implementations. you need to remove it here too.

File: src/Jackalope/Transport/NodeTypeFilterInterface.php [hunk 11]
```diff
@@ -30,7 +30,7 @@
      *
      * @see TransportInterface::getNodes
      */
-    public function getNodesFiltered($paths, $typeFilter);
+    public function getNodesFiltered(array $paths, $typeFilter);
```
Comment (dbu) [inline src/Jackalope/Transport/NodeTypeFilterInterface.php]: this is a BC break, please revert

File: src/Jackalope/Transport/TransportInterface.php [hunk 12]
```diff
@@ -319,12 +324,11 @@ public function setNodeTypeManager($nodeTypeManager);
...
      *
-     * @return array with the definitions (see
-     *      Jackalope\NodeTypeDefinition::fromArray for what is expected)
+     * @return array with the definitions
      *
-     * @see Jackalope\NodeTypeDefinition::fromArray
+     * @see NodeTypeDefinition::fromArray
      */
-    public function getNodeTypes($nodeTypes = array());
+    public function getNodeTypes(array $nodeTypes = array());
```
Comment (dbu) [inline src/Jackalope/Transport/TransportInterface.php]: this is a BC break. please rebase on the upstream master, where i reverted this.

**Comment by vondrasekpavel** [comment]
Oh, sorry :( 

For now I will revert this change. I will try to fix it later.

File: src/Jackalope/ImportExport/ImportExport.php [hunk 13]
```diff
@@ -464,10 +502,10 @@ private static function importSystemView(NodeInterface $parentNode, NamespaceReg
         if (! $xml->read()) {
             throw new InvalidSerializedDataException('missing information to create node');
         }
-        if ('property' != $xml->localName || NamespaceRegistryInterface::NAMESPACE_SV != $xml->namespaceURI) {
+        if ('property' !== $xml->localName || NamespaceRegistryInterface::NAMESPACE_SV !== $xml->namespaceURI) {
             throw new InvalidSerializedDataException('first child of node must be sv:property for jcr:primaryType. Found {'.$xml->namespaceURI.'}'.$xml->localName.'="'.$xml->value.'"'.$xml->nodeType);
         }
-        if (! $xml->moveToAttributeNs('name', NamespaceRegistryInterface::NAMESPACE_SV) || 'jcr:primaryType' != $xml->value) {
+        if ('jcr:primaryType' !== $xml->value || !$xml->moveToAttributeNs('name', NamespaceRegistryInterface::NAMESPACE_SV)) {
```
Comment (dbu) [inline src/Jackalope/ImportExport/ImportExport.php]: i tested with jackalope-doctrine-dbal, and this line is a regression. the problem is that we call moveToAttributeNs for its side effect of moving. plus, the check for value must happen after we moved to the attribute, not before.

to make this more explicit, we could split this in two like this:

```
if (! $xml->moveToAttributeNs('name', NamespaceRegistryInterface::NAMESPACE_SV)) {
    throw new InvalidSerializedDataException('first child of node must have a sv:name property');
}
if ('jcr:primaryType' !== $xml->value) {
    throw new InvalidSerializedDataException('first child of node must be the sv:property node with a jcr:primaryType. Found {'.$xml->namespaceURI.'}'.$xml->localName.'="'.$xml->value
}
```

do you want to update like this? i think if you change that, this would be ready to merge.

(if you want, you could search this file for " == " - i found some places where we don't use === that you missed.

**Comment by dbu** [comment]
please rebase on master, github sees a conflict.

if you want, you could search this file for " == " - i found some places where we don't use === that you missed.

File: src/Jackalope/ImportExport/ImportExport.php (hunk 13 above)
Comment (vondrasekpavel) [inline src/Jackalope/ImportExport/ImportExport.php]: ok, I will :)

**Comment by dbu** [comment]
great, tests with jackalope-doctrine-dbal are now green. can you please squash the commits?

**Comment by dbu** [comment]
squashed and merged manually in 5f59291a191a29fab16c124ffe725fbe3976f823

thanks a lot for this cleanup work @vondrasekpavel !
